src/main/java/uk/gov/hmcts/reform/sendletter/blob/component/BlobReader.java

```diff
@@ -1,29 +1,49 @@
 package uk.gov.hmcts.reform.sendletter.blob.component;
 
+import com.azure.storage.blob.BlobContainerClient;
 import com.azure.storage.blob.models.BlobItem;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sendletter.config.AccessTokenProperties;
+import uk.gov.hmcts.reform.sendletter.model.in.ManifestBlobInfo;
+
+import java.util.ArrayList;
+import java.util.Optional;
 
 @Component
 public class BlobReader {
 
     private static final Logger LOG = LoggerFactory.getLogger(BlobReader.class);
 
     private final BlobManager blobManager;
-    private static final String NEW_CONTAINER = "new";
+    private final AccessTokenProperties accessTokenProperties;
 
-    public BlobReader(BlobManager blobManager) {
+    public BlobReader(BlobManager blobManager, AccessTokenProperties accessTokenProperties) {
         this.blobManager =  blobManager;
+        this.accessTokenProperties = accessTokenProperties;
     }
 
-    public String retrieveFileToProcess() {
-        LOG.info("About to read new blob connection details");
-        var containerClient = blobManager.getContainerClient(NEW_CONTAINER);
-        String blobName = containerClient.listBlobs().stream().findFirst().map(BlobItem::getName).orElse("");
-        LOG.info("BlobReader:: Blob name: {}", blobName);
-        return blobName;
-    }
+    public Optional<ManifestBlobInfo> retrieveManifestsToProcess() {
+        LOG.info("About to read manifests details");
+        var containers = new ArrayList<>(accessTokenProperties.getServiceConfig());
+
+        for (AccessTokenProperties.TokenConfig config : containers) {
```

Comment:
i think we should randomise lookup of manifest file or else we would pick manifest file from the same service.  This could be done in the next upgrade

Reply:
Added.